cmd/vicadmin/logs.go

```diff
@@ -302,55 +304,105 @@ func zipEntries(readers map[string]entryReader, out *zip.Writer) error {
 }
 
 func tailFile(wr io.Writer, file string, done *chan bool) error {
-	nlines := tailLines
-	f, err := os.Open(file)
-	if err != nil {
-		return err
-	}
+	defer trace.End(trace.Begin(""))
 
-	pos, err := f.Seek(0, 2)
-	if err != nil {
-		return err
+	// By default, seek to EOF (if file doesn't exist)
+	spos :=   tail.SeekInfo{
+		Offset: 0,
+		Whence: 2,
 	}
-	b := make([]byte, 1)
-	for pos > 0 && nlines >= 0 {
-		pos--
-		_, err := f.ReadAt(b, pos)
-		if err != nil && err != io.EOF {
-			return err
-		}
-		if b[0] == '\n' {
-			nlines--
+
+	// If the file exists, we want to go back tailLines lines
+	// and pass that new offset into the TailFile() constructor
+	// Per @fdawg4l, use bytes.LastIndex() and a 1k buffer to reduce
+	// seeks/reads
+	f, err := os.Open(file)
+	if err == nil {
```

Comment:
There is also this: https://github.com/landley/toybox/blob/master/toys/posix/tail.c#L95